src/components/DisplayDataTable/index.tsx

```diff
@@ -0,0 +1,330 @@
+        {tableTitle && (
+          <Text
+            marginRight="14px"
+            fontSize="24px"
+            fontWeight="700"
+            color="#000000"
+            data-testid="table-title"
+          >
+            {tableTitle}
+          </Text>
+        )}
+        {showTotalQuantity && (
+          <Badge
+            padding="4px 12px"
+            borderRadius="25px"
+            backgroundColor="#2C4E8A"
+            color="#FFFFFF"
+            data-testid="total-count"
+          >
+            {totalCount}
+          </Badge>
+        )}
+      </Flex>
+
+      {/* Search / Filter Button / Create Button */}
+      {(tableSearch || createLink) && (
+        <Flex
+          marginTop="16px"
+          justifyContent={tableSearch !== undefined ? "space-between" : "flex-end"}
+        >
+          {tableSearch && (
+            <Box width="fit-content" data-testid="table-search">
+              <Search
+                searchPlaceholder={tableSearch.searchPlaceholder}
+                searchValue={tableSearch.searchValue as string}
+                setSearchValue={tableSearch.setSearchValue as Dispatch<SetStateAction<string>>}
+              ></Search>
+            </Box>
+          )}
+          <Flex>
+            {/* Todo: filter function */}
+            {/* <Button
+              leftIcon={<HiViewGridAdd />}
+              backgroundColor="#2C4E8A"
+              color="#FFFFFF"
+              _hover={{
+                backgroundColor: "#7088B1",
+                color: "#F3F2F7",
+              }}
+            >
+              Add Filter
+            </Button> */}
+            {createLink && (
+              <Button
+                leftIcon={<HiPlus />}
+                marginLeft="10px"
+                backgroundColor="#2C4E8A"
+                color="#FFFFFF"
+                _hover={{
+                  backgroundColor: "#7088B1",
+                  color: "#F3F2F7",
+                }}
+                onClick={() => {
+                  router.push(createLink as string);
+                }}
+                data-testid="create-button"
+              >
+                Create
+              </Button>
+            )}
+          </Flex>
+        </Flex>
+      )}
+
+      {/* Table */}
+      {tableSearch?.searchLoading ? (
+        <Center height="100vh">
+          <Loading loadingText={tableSearch?.searchLoadingText}></Loading>
+        </Center>
+      ) : totalCount ? (
+        <>
+          <Table marginTop="24px" variant="striped">
+            <Thead backgroundColor="#2C4E8A">
+              {table.getHeaderGroups().map((headerGroup) => (
+                <Tr key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => {
+                    return (
+                      <Th
+                        key={header.id}
+                        fontSize="12px"
+                        color="#FFFFFF"
+                        cursor={header.column.getCanSort() ? "pointer" : "default"}
```

Comment:
when clicking the header's word, the position of the word will change, and in my opinion, it should have a sort related to time, update tim
![image](https://user-images.githubusercontent.com/106716226/201355208-26bfe02c-f73d-401b-b5e0-81418ae04c9a.png)
e or create time. and not sure the reason quotation and quotation detail no sort function.

Reply:
let's discuss when you have time